### PR TITLE
Improve naming and asserts of autoscale test

### PR DESCRIPTION
**Pull request by dafeda**
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure unit tests pass locally after every commit (`git rebase -i main
      --exec 'just rapid-tests'`)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Add backport label to latest release (format: 'backport release-branch-name')

### CODE REVIEW

File: tests/ert/unit_tests/analysis/test_misfit_preprocessor.py [hunk 1]
```diff
@@ -142,18 +142,29 @@ def test_that_correlated_and_independent_observations_are_grouped_separately(
 
 
 @pytest.mark.parametrize("nr_observations", [0, 1, 2])
-def test_misfit_preprocessor_single(nr_observations):
-    """We create nr_observations responses using the linear function y = ax.
-    We don`t have enough information to cluster properly, so we don`t try
-    to scale"""
-    rng = np.random.default_rng(1234)
+def test_that_perfectly_correlated_responses_are_not_scaled(nr_observations):
```
Comment (xjules) [inline tests/ert/unit_tests/analysis/test_misfit_preprocessor.py]: `test_that_responses_with_correlation_one_are_not_scaled`?

Comment (dafeda) [reply tests/ert/unit_tests/analysis/test_misfit_preprocessor.py]: Thanks, but I prefer the original :)

Comment (larsevj) [reply tests/ert/unit_tests/analysis/test_misfit_preprocessor.py]: I think the idea behind this test is just to test the following code path in misfit preprocessor:
```
    if len(obs_errors) <= 2:
        logger.info("Observations not correlated or only correlated each other")
        return scale_factors, np.ones(len(obs_errors), dtype=int), nr_components
```
because this test will fail if you increase the number of observations beyond 2, as then the correlated group will be scaled as expected.

Comment (larsevj) [reply tests/ert/unit_tests/analysis/test_misfit_preprocessor.py]: Now why we do not want to scale in the case that you only input 2 observations, I am not entirely sure.

Comment (dafeda) [reply tests/ert/unit_tests/analysis/test_misfit_preprocessor.py]: Nice catch Lars!
Should we just rename the test and leave as is for now?

Comment (larsevj) [reply tests/ert/unit_tests/analysis/test_misfit_preprocessor.py]: I would say rename test and keep the doc-string ( or improve it a bit), but the additional asserts cannot hurt to have as well.

Comment (dafeda) [reply tests/ert/unit_tests/analysis/test_misfit_preprocessor.py]: I have changed the name of the test to `test_edge_cases_with_few_observations_return_default_values`.

File: tests/ert/unit_tests/analysis/test_misfit_preprocessor.py [hunk 2]
```diff
@@ -142,18 +142,32 @@ def test_that_correlated_and_independent_observations_are_grouped_separately(
...
 
 @pytest.mark.parametrize("nr_observations", [0, 1, 2])
-def test_misfit_preprocessor_single(nr_observations):
-    """We create nr_observations responses using the linear function y = ax.
-    We don`t have enough information to cluster properly, so we don`t try
-    to scale"""
-    rng = np.random.default_rng(1234)
+def test_edge_cases_with_few_observations_return_default_values(nr_observations):
+    """Test that edge cases with 0-2 observations return default scaling values.
+    We do not know why this is the case.
```
Comment (larsevj) [inline tests/ert/unit_tests/analysis/test_misfit_preprocessor.py:147]: I guess it trivially makes sense for 0 and 1, but not 2.
